snaps/promote-snaps.sh

```diff
@@ -6,19 +6,39 @@ set -eu
 # export PROMOTE_TO="1.10/beta 1.10/candidate edge beta candidate"
 # snaps/promote-snaps.sh
 
+ARCH=${KUBE_ARCH:-"amd64"}
+GH_USER="${GH_USER:-}"
+GH_TOKEN="${GH_TOKEN:-}"
 SNAPS="${SNAPS:-kubectl kube-apiserver kube-controller-manager kube-scheduler kubelet kube-proxy cdk-addons kubeadm kubernetes-test}"
 
-ARCH=${KUBE_ARCH:-"amd64"}
 echo PROMOTE_FROM="$PROMOTE_FROM"
 echo PROMOTE_TO="$PROMOTE_TO"
 echo SNAPS="$SNAPS"
 echo ARCH="$ARCH"
 
-
 . utils/retry.sh
 
-GH_USER="${GH_USER:-}"
-GH_TOKEN="${GH_TOKEN:-}"
+# Set a flag if we're building the EKS variants
+if [[ "${SNAPS}" =~ "-eks" ]]; then
+  build_eks=true
+fi
+
+# Ensure we are the correct user
+if [ "$build_eks" = true ]; then
+  cp -a /var/lib/jenkins/.config/snapcraft/snapcraft-cpc.cfg \
```

Comment:
Meh, i don't like this anymore.  This opens us up for problems with concurrent jobs trying to twiddle the filesystem at the same time.